src/login/logind-session-device.c

```diff
@@ -478,3 +479,24 @@ unsigned int session_device_try_pause_all(Session *s) {
 
         return num_pending;
 }
+
+int session_restore_devices(Session *s, const char *devices) {
+        const char *word, *state;
+        size_t l;
+        int r;
+
+        assert(s);
+
+        if (devices)
+                FOREACH_WORD(word, l, devices, state) {
+                        SessionDevice *sd;
+                        dev_t dev = strtoul(word, NULL, 10);
+
+                        r = session_device_new(s, dev, &sd);
+                        if (r < 0)
+                                log_warning("Failed to restore device [%u:%u]: %m",
+                                            major(dev), minor(dev));
```

Comment:
as discussed here already by others: the only way this can work correctly is if we keep the fds open across the restart. There's infrastructure for that in place now using the fdstore stuff. It's pretty easy to use even too.